odbc_fdw.c

```diff
@@ -844,29 +1070,16 @@ static void odbcGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid fo
 static void odbcEstimateCosts(PlannerInfo *root, RelOptInfo *baserel, Cost *startup_cost, Cost *total_cost, Oid foreigntableid)
 {
 	unsigned int table_size   = 0;
-	char *svr_dsn             = NULL;
-	char *svr_driver          = NULL;
-	char *svr_host            = NULL;
-	char *svr_port            = NULL;
-	char *svr_database        = NULL;
-	char *svr_schema          = NULL;
-	char *svr_table           = NULL;
-	char *sql_query           = NULL;
-	char *sql_count           = NULL;
-	char *username            = NULL;
-	char *password            = NULL;
-	List *col_mapping_list;
+	odbcFdwOptions options;
```

Comment:
I've left initialization to the function that extracts fills the options structure (so `extract_odbcFdwOptions` is the one doing the initialization). So, now functions such as `odbcGetTableOptions` which is used here to fill the options admit an uninitialized options structure.

The options structure should probably be extended to store and open set of odbc connection attributes, then it might be better to avoid storing it in the stack and hava a function to allocate and initialize it in the heap, so we'd replace the declaration there by something like:

```
odbcFdwOptions* options = new_odbcFdwOption()
```